x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { CredentialsLogic } from './credentials_logic';
+import { ADMIN, PRIVATE } from '../../constants/credentials';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn(), delete: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+import { IApiToken, ICredentialsDetails } from '../../../../../common/types/app_search';
+import { IMeta } from '../../../../../common/types';
+
+function mount(defaults?: object) {
+  if (!defaults) {
+    resetContext({});
+  } else {
+    resetContext({
+      // the enterprise_search.app_search.credentials namespace is set in `path` in credentials_logic
+      defaults: {
+        enterprise_search: {
+          app_search: {
+            credentials: {
+              ...defaults,
+            },
+          },
+        },
+      },
+    });
+  }
+  CredentialsLogic.mount();
+}
+
+const newToken: IApiToken = {
+  id: 1,
+  name: 'myToken',
+  type: PRIVATE,
+  read: true,
+  write: true,
+  access_all_engines: true,
+};
+
+const credentialsDetails: ICredentialsDetails = {
+  lmAccount: {
+    id: '1',
+    key: '1234',
+  },
+  apiTokens: [],
+  apiUrl: 'http://example.com',
+  engines: [
+    { name: 'engine1', type: 'indexed', language: 'english', result_fields: [] },
+    { name: 'engine1', type: 'indexed', language: 'english', result_fields: [] },
+  ],
+};
```

Comment:
If any other files use this at all, it may potentially be worth moving these into a `components/credentials/__mocks__/index.ts` file. It may be worth that right now TBH, since this is such a long file, and it'll help readability to shift setup/allow devs to dive into the actual tests sooner

Reply:
It's a good thought. The `credentialsDetails` is quite a bit shorter after making some updates here: https://github.com/elastic/kibana/commit/d1c2444c82b8cd0ac74afbdf583bc1c3a3654379.

I also just wonder if we start pulling out all of these little data values from tests if our mocks directory is going to explode in size.

I think this could be worth it if we end up using these in component tests as well. I think I'll wait until then.